### PR TITLE
Add project documentation and README

### PR DESCRIPTION
**Pull request by darshan-lukhi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,127 @@
...
+| Feature                         | Supported | Notes |
+| ------------------------------- | --------- | ----- |
+| Native Queries + Logical Models | ❌         |       |
+| Simple Object Query             | ✅         |       |
+| Filter / Search                 | ✅         |       |
+| Simple Aggregation              | ✅         |       |
+| Sort                            | ✅         |       |
+| Paginate                        | ✅         |       |
+| Nested Objects                  | ✅         |       |
+| Nested Arrays                   | ✅         |       |
```
Comment (gneeri) [inline README.md:36]: I believe we have to remove nested filtering and sorting, until this is supported by the engine. We can update this when that comes online (hopefully, in the next few weeks).

Comment (darshan-lukhi) [reply README.md:33]: @gneeri Agree, Removed nested filtering and sorting

File: README.md [hunk 2]
```diff
@@ -1,48 +1,124 @@
...
+# Elasticsearch Connector
+
+<a href="https://hasura.io/"><img src="./docs/logo.png" align="right" width="200"></a>
+
+[![Docs](https://img.shields.io/badge/docs-v3.x-brightgreen.svg?style=flat)](https://hasura.io/docs/3.0/latest/connectors/elasticsearch/)
+[![ndc-hub](https://img.shields.io/badge/ndc--hub-elasticsearch-blue.svg?style=flat)](https://hasura.io/connectors/ndc-elasticsearch)
+[![License](https://img.shields.io/badge/license-Apache--2.0-purple.svg?style=flat)](LICENSE.txt)
+[![Status](https://img.shields.io/badge/status-alpha-yellow.svg?style=flat)](./readme.md)
+
+The Hasura Elasticsearch Connector allows for connecting to a Elasticsearch search engine, giving you an instant
```
Comment (rsd1122) [inline README.md:10]:
```suggestion
With this connector, Hasura allows you to instantly create a real-time GraphQL API on top of your documents in Elasticsearch. This connector supports Elasticsearch functionalities listed in the table below, allowing for efficient and scalable data operations. Additionally, you will benefit from all the powerful features of Hasura’s Data Delivery Network (DDN) platform, including query pushdown capabilities that delegate all query operations to the Elasticsearch, thereby enhancing query optimization and performance.

```

File: README.md [hunk 3]
```diff
@@ -1,48 +1,124 @@
...
+GraphQL API on top of your Elasticsearch data.
```
Comment (rsd1122) [inline README.md:11]:
```suggestion
```

File: docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# General Architecture of the Elasticsearch Connector
+
+### Overview
+
+The Elasticsearch Connector takes a `QueryRequest`, which contains information about the query a user would like to run, translates it to an Elasticsearch DSL (Domain Specific Language) query, executes it against Elasticsearch, and returns the results as a `QueryResponse`.
+
+### Components
+
+The architecture of the Elasticsearch Connector is composed of three main components:
```
Comment (rsd1122) [inline docs/architecture.md:9]:
```suggestion
The connector has three main components:
```

File: docs/architecture.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  state *types.State
+) (map[string]interface{}, error)
+```
+
+#### 2. Search
+
+Executes the DSL query against Elasticsearch and returns results.
+
+**Functionality:**
+- Execute the query on the Elasticsearch index.
```
Comment (rsd1122) [inline docs/architecture.md:40]:
```suggestion
- Execute queries on the specified Elasticsearch index.
```

File: docs/architecture.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+- Fetch raw search results.
```
Comment (rsd1122) [inline docs/architecture.md:41]:
```suggestion
- Fetch search results
```

File: docs/architecture.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  body map[string]interface{}
+) (map[string]interface{}, error)
+```
+
+#### 3. prepareResponse
+
+Converts Elasticsearch search results into `QueryResponse`.
+
+**Functionality:**
+- Traverse through each field in the Elasticsearch response.
```
Comment (rsd1122) [inline docs/architecture.md:58]:
```suggestion
- Traverse each field in the Elasticsearch response.
```

File: docs/architecture.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+- Construct and return the `QueryResponse`, ensuring it aligns with the expected fields.
```
Comment (rsd1122) [inline docs/architecture.md:59]:
```suggestion
- Construct and return the QueryResponse, ensuring it conforms to the expected fields
```

**Pull request merged**